### ISSUE TITLE
Submodules from git

### ISSUE THREAD

**Issue by fattikus** [neurospeech/iis-ci#1]
Hey,

So I was trying to start using this IIS-CI however I had an issue because my git uses submodules. Problem is submodules are not being downloaded even tought i changed the GitSourceController like this:

`if (!Directory.Exists(gitFolder))
            {
                Directory.CreateDirectory(gitFolder);

                Console.WriteLine("Cloning repository " + config.SourceUrl);

                CloneOptions clone = new CloneOptions();
                clone.CredentialsProvider = CredentialsHandler;
                clone.RecurseSubmodules = true;
                
                Repository.Clone(config.SourceUrl, gitFolder, clone);

                Console.WriteLine("Repository clone successful");
            }`

Do you know how I can get IIS-CI to get git to recurse submodules? the git command would be --recurse-submodules=no origin
Notice that above I added a line to clone, so now it also has clone.RecurseSubmodules
Edit: For pulling before building

Could this be a limitation on libgit2sharp

**Comment by neurospeech** [comment]
Yes this looks as a bug in libgit2sharp, in that case we are trying to find an alternative to use git.exe,
